### PR TITLE
Feat/pando 19

### PR DESCRIPTION
**Pull request by sssion**
update ut to convey and gomonkey framework

### CODE REVIEW

File: api/v0/admin/model/register_request_test.go [hunk 1]
```diff
@@ -1,46 +1,60 @@
...
-	}
-
-	if seq0 >= peerRec.Seq {
-		t.Fatal("sequence not greater than last seen")
-	}
+	addrs := []string{"/ip4/127.0.0.1/tcp/9999"}
+	Convey("test create and load register request", t, func() {
+		So(err, ShouldBeNil)
+		data, err := MakeRegisterRequest(peerID, privKey, addrs, account)
+		So(err, ShouldBeNil)
```
Comment (bsjohnson01) [inline api/v0/admin/model/register_request_test.go:22]: I think the data should be read then verified if it equals to the args you just passed to the request-maker

File: api/v0/admin/model/register_request_test.go [hunk 2]
```diff
@@ -1,46 +1,60 @@
...
+		peerRec, err := ReadRegisterRequest(data)
+		So(err, ShouldBeNil)
+		seq0 := peerRec.Seq
+		// register again
```
Comment (bsjohnson01) [inline api/v0/admin/model/register_request_test.go:26]: What is the purpose of re-registering here?

File: api/v0/admin/model/register_request_test.go [hunk 3]
```diff
@@ -1,46 +1,60 @@
...
+		// seq create from time
+		So(seq0, ShouldBeLessThan, peerRec.Seq)
+
+	})
+	Convey("test create error and load error", t, func() {
+		Convey("nil addr", func() {
+			account := "t12345"
+			peerID, privKey, err := mock.GetPrivkyAndPeerID()
+			So(err, ShouldBeNil)
+			_, err = MakeRegisterRequest(peerID, privKey, []string{}, account)
```
Comment (bsjohnson01) [inline api/v0/admin/model/register_request_test.go]: Maybe it's necessary to ensure the returned data should be nil if a non-nil err returned

File: config/identity_test.go [hunk 4]
```diff
@@ -2,20 +2,21 @@ package config
...
-	}
-	pk, err := id.DecodePrivateKey("")
-	if err != nil {
-		t.Fatal(err)
-	}
-	if pk.Type() != crypto_pb.KeyType_Ed25519 {
-		t.Fatal("unexpected type:", pk.Type())
-	}
+	Convey("test create and decode identity", t, func() {
+		identity, err := CreateIdentity(io.Discard)
```
Comment (bsjohnson01) [inline config/identity_test.go:12]: To test the CreateIdentity function, you may compare the identity which is the actual output of this function with your expect output

**Review by bsjohnson01 — CHANGES_REQUESTED** [review]

File: api/v0/admin/model/register_request_test.go (hunk 1 above)
Comment (sssion) [reply api/v0/admin/model/register_request_test.go:22]: 返回的是一个[]bytes,  我如果加以判断的话可能就是 判断 他生成的[]byte是否等于事先用他生成的[]byte

File: api/v0/admin/model/register_request_test.go (hunk 2 above)
Comment (sssion) [reply api/v0/admin/model/register_request_test.go:26]: 不应该有error，  只要timestamp正常

File: config/identity_test.go (hunk 4 above)
Comment (sssion) [reply config/identity_test.go:12]: 这个func 生成random....

**Pull request merged**